src/Rules/PhpDoc/IncompatiblePhpDocTypeRule.php

```diff
@@ -137,6 +141,24 @@ public function processNode(Node $node, Scope $scope): array
 						),
 					));
 
+					TypeTraverser::map($phpDocParamType, function (Type $type, callable $traverse) use ($escapedTagName, $escapedParameterName, $node, $scope, &$errors, $functionName, $resolvedPhpDoc): Type {
+						if ($type instanceof CallableType || $type instanceof ClosureType) {
+							$classReflection = $scope->isInClass() ? $scope->getClassReflection() : null;
+							$errors = array_merge($errors, $this->genericCallableRuleHelper->check(
```

Comment:
This could just be this with the TypeTraverser call done inside, right?

```php

							$errors = array_merge($errors, $this->genericCallableRuleHelper->check(
								$node,
								$scope,
								sprintf('%s for parameter $%s', $escapedTagName, $escapedParameterName),
								$type,
								$functionName,
								$resolvedPhpDoc->getTemplateTags(),
								$classReflection,
								$classReflection !== null ? $classReflection->getTemplateTags() : [],
							));
```

Reply:
yeah it can, short sighted of me :sweat_smile: